From d6017aa0da2340f8d87bf649cefbceede58fff96 Mon Sep 17 00:00:00 2001
From: Amir Ghanbari Bavarsad <amir.b@pervices.com>
Date: Tue, 26 Mar 2019 16:10:46 +0000
Subject: [PATCH 8/9] Delay RGMII RX data by 1.38 ns relative to clock

It seems that hold timing is still being violated in the FPGA. To help
with HOLD, we need to send the clock sooner.
---
 .../dts/altera/socfpga_stratix10_socdk.dts    | 24 +++++++++----------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
index 6635022dbb2a..4b84aa7d8b81 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk.dts
@@ -90,12 +90,12 @@
 			txen-skew-ps = <0>; /* -420ps */
 			txc-skew-ps  = <1860>; /* 960ps */
 
-			rxd0-skew-ps = <420>; /* 0ps */
-			rxd1-skew-ps = <420>; /* 0ps */
-			rxd2-skew-ps = <420>; /* 0ps */
-			rxd3-skew-ps = <420>; /* 0ps */
-			rxdv-skew-ps = <420>; /* 0ps */
-			rxc-skew-ps  = <900>; /* 0ps */
+			rxd0-skew-ps = <900>; /*  480ps */
+			rxd1-skew-ps = <900>; /*  480ps */
+			rxd2-skew-ps = <900>; /*  480ps */
+			rxd3-skew-ps = <900>; /*  480ps */
+			rxdv-skew-ps = <900>; /*  480ps */
+			rxc-skew-ps  = <0>;   /* -900ps */
 		};
 	};
 };
@@ -132,12 +132,12 @@
 			txen-skew-ps = <0>; /* -420ps */
 			txc-skew-ps  = <1860>; /* 960ps */
 
-			rxd0-skew-ps = <420>; /* 0ps */
-			rxd1-skew-ps = <420>; /* 0ps */
-			rxd2-skew-ps = <420>; /* 0ps */
-			rxd3-skew-ps = <420>; /* 0ps */
-			rxdv-skew-ps = <420>; /* 0ps */
-			rxc-skew-ps  = <900>; /* 0ps */
+			rxd0-skew-ps = <900>; /*  480ps */
+			rxd1-skew-ps = <900>; /*  480ps */
+			rxd2-skew-ps = <900>; /*  480ps */
+			rxd3-skew-ps = <900>; /*  480ps */
+			rxdv-skew-ps = <900>; /*  480ps */
+			rxc-skew-ps  = <0>;   /* -900ps */
 		};
 	};
 };
-- 
2.19.1

